zokrates_core/src/absy/mod.rs

```diff
@@ -44,61 +42,123 @@ pub struct Program<'ast> {
+    ) -> Self {
+        SymbolImport {
+            symbol_id: symbol_id.into(),
+            module_id: module_id.into(),
+        }
+    }
+}
+
+impl<'ast> fmt::Display for SymbolImport<'ast> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "from \"{}\" import {}",
+            self.module_id.display(),
+            self.symbol_id
+        )
+    }
+}
+
+/// A declaration of a symbol
+#[derive(Debug, PartialEq, Clone)]
 pub struct SymbolDeclaration<'ast> {
     pub id: Identifier<'ast>,
     pub symbol: Symbol<'ast>,
 }
 
 #[allow(clippy::large_enum_variant)]
-#[derive(PartialEq, Clone)]
+#[derive(Debug, PartialEq, Clone)]
 pub enum SymbolDefinition<'ast> {
+    Import(CanonicalImportNode<'ast>),
     Struct(StructDefinitionNode<'ast>),
     Constant(ConstantDefinitionNode<'ast>),
     Function(FunctionNode<'ast>),
 }
 
-impl<'ast> fmt::Debug for SymbolDefinition<'ast> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            SymbolDefinition::Struct(s) => write!(f, "Struct({:?})", s),
-            SymbolDefinition::Constant(c) => write!(f, "Constant({:?})", c),
-            SymbolDefinition::Function(func) => write!(f, "Function({:?})", func),
-        }
-    }
-}
-
-#[derive(PartialEq, Clone)]
+#[derive(Debug, PartialEq, Clone)]
 pub enum Symbol<'ast> {
     Here(SymbolDefinition<'ast>),
     There(SymbolImportNode<'ast>),
     Flat(FlatEmbed),
 }
 
-impl<'ast> fmt::Debug for Symbol<'ast> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            Symbol::Here(k) => write!(f, "Here({:?})", k),
-            Symbol::There(i) => write!(f, "There({:?})", i),
-            Symbol::Flat(flat) => write!(f, "Flat({:?})", flat),
-        }
-    }
-}
-
 impl<'ast> fmt::Display for SymbolDeclaration<'ast> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self.symbol {
-            Symbol::Here(ref kind) => match kind {
-                SymbolDefinition::Struct(t) => write!(f, "struct {} {}", self.id, t),
-                SymbolDefinition::Constant(c) => write!(
+        match &self.symbol {
+            Symbol::Here(ref symbol) => match symbol {
+                SymbolDefinition::Import(ref i) => write!(f, "{}", i),
+                SymbolDefinition::Struct(ref t) => write!(f, "struct {} {}", self.id, t),
+                SymbolDefinition::Constant(ref c) => write!(
                     f,
                     "const {} {} = {}",
                     c.value.ty, self.id, c.value.expression
                 ),
-                SymbolDefinition::Function(func) => write!(f, "def {}{}", self.id, func),
+                SymbolDefinition::Function(ref func) => {
+                    write!(f, "def {}{}", self.id, func)
+                }
             },
-            Symbol::There(ref import) => write!(f, "import {} as {}", import, self.id),
+            Symbol::There(ref i) => write!(f, "{}", i),
```

Comment:
Am I right that we end up with some information duplication here? In the other match arms, we use `self.id` which is the name of the symbol in this module. This makes me wonder if `i` here doesn't have information it doesn't need? Maybe the alias needs to be kept out of `i` so we only rely on `self.id`?